src/shogun/neuralnets/StanNeuralNetwork.h

```diff
@@ -0,0 +1,195 @@
+ * input layer. The last layer in the array is treated as the output layer.
+ * Also note that forward propagation is performed in the order at which the
+ * layers appear in the array. So if layer j takes its input from layer i then
+ * i must be less than j.
+ * 	- Specify how the layers are connected together. This can be done using
+ * either connect() or quick_connect().
+ * 	- Call initialize_neural_network()
+ *  - Train using an optimizer
+ * 	- Apply the network using apply()
+ *
+ * Supported feature types: CDenseFeatures<float64_t>
+ * Supported label types:
+ * 	- CBinaryLabels
+ * 	- CMulticlassLabels
+ * 	- CRegressionLabels
+ *
+ * The neural network can be trained using any of the minimizers in the optimization module
+ *
+ * The network stores the parameters of all the  layers in a single array. This
+ * makes it easy to train a network of any combination of arbitrary layer types
+ * using any optimization method
+ *
+ * All the matrices the network (and related classes) deal with are in
+ * column-major format
+ *
+ */
+class StanNeuralNetwork : public CSGObject
+{
+
+public:
+	/** default constuctor */
+	StanNeuralNetwork();
+
+	/** Sets the layers of the network
+	 *
+	 * @param layers An array of CNeuralLayer objects specifying the layers of
+	 * the network. Must contain at least one input layer. The last layer in
+	 * the array is treated as the output layer
+	 */
+	StanNeuralNetwork(CDynamicObjectArray* layers);
+
+	/** Sets the layers of the network
+	 *
+	 * @param layers An array of CNeuralLayer objects specifying the layers of
+	 * the network. Must contain at least one input layer. The last layer in
+	 * the array is treated as the output layer
+	 */
+	virtual void set_layers(CDynamicObjectArray* layers);
+
+	/** Connects layer i as input to layer j. In order for forward and
+	 * backpropagation to work correctly, i must be less that j
+	 */
+	virtual void connect(int32_t i, int32_t j);
+
+	/** Connects each layer to the layer after it. That is, connects layer i to
+	 * as input to layer i+1 for all i.
+	 */
+	virtual void quick_connect();
+
+	/** Disconnects layer i from layer j */
+	virtual void disconnect(int32_t i, int32_t j);
+
+	/** Removes all connections in the network */
+	virtual void disconnect_all();
+
+	/** Initializes the network
+	 *
+	 * @param sigma standard deviation of the gaussian used to randomly
+	 * initialize the parameters
+	 */
+	virtual void initialize_neural_network(float64_t sigma = 0.01f);
+
+	virtual ~CNeuralNetwork();
+
+	/** apply machine to data in means of binary classification problem */
+	virtual CBinaryLabels* apply_binary(CFeatures* data);
+	/** apply machine to data in means of regression problem */
+	virtual CRegressionLabels* apply_regression(CFeatures* data);
+	/** apply machine to data in means of multiclass classification problem */
+	virtual CMulticlassLabels* apply_multiclass(CFeatures* data);
```

Comment:
I am not sure these ones should be here. Let's check with @karlnapf or @vigsterkr.